kratos/utilities/dense_svd_decomposition.h

```diff
@@ -0,0 +1,188 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_DENSE_SVD_H_INCLUDED)
+#define KRATOS_DENSE_SVD_H_INCLUDED
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+
+namespace Kratos {
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template<class TDenseSpaceType>
+class DenseSingularValueDecomposition
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Definition of the shared pointer of the class
+    KRATOS_CLASS_POINTER_DEFINITION(DenseSingularValueDecomposition);
+
+    typedef typename TDenseSpaceType::DataType DataType;
+    typedef typename TDenseSpaceType::VectorType VectorType;
+    typedef typename TDenseSpaceType::MatrixType MatrixType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    DenseSingularValueDecomposition() = default;
+
+    virtual ~DenseSingularValueDecomposition() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Name of the SVD
+     * Returns a string containing the name of the current SVD decomposition
+     * @return std::string Name of the SVD decomposition
+     */
+    static std::string Name()
+    {
+        return "dense_singular_value_decomposition";
+    }
+
+    /**
+     * @brief Compute the SVD
+     * Computes the Singular Value Decomposition (SVD) of the given imput matrix
+     * @param rInputMatrix Matrix to compute the SVD decomposition
+     * @param Settings Settings for the SVD decomposition
+     */
+    virtual void Compute(
+        MatrixType& rInputMatrix,
+        Parameters Settings) = 0;
+
+    /**
+     * @brief Compute the SVD
+     * Computes the Singular Value Decomposition (SVD) of the given input matrix
+     * @param rInputMatrix Matrix to compute the SVD decomposition
+     * @param rVectorS Vector containing the singular values (sorted from the largest one to smallest one)
+     * @param rMatrixU Left singular vectors matrix
+     * @param rMatrixV Right singular vectors matrix
+     * @param Settings Settings for the SVD decomposition
+     */
+    virtual void Compute(
+        MatrixType& rInputMatrix,
```

Comment:
const?

Reply:
Can't be const as the `Eigen::Map` doesn't allowed (I already tried).